substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+		pub fn on_registration(origin: OriginFor<T>, project_id: ProjectId<T>) -> DispatchResult {
+			let _who = ensure_signed(origin.clone())?;
+			let mut infos = WhiteListedProjectAccounts::<T>::get(project_id.clone())
+				.ok_or(Error::<T>::NoProjectAvailable)?;
+
+			let round_index = NextVotingRoundNumber::<T>::get();
+			let current_round_index = round_index.saturating_sub(1);
+			let round_infos =
+				VotingRounds::<T>::get(current_round_index).ok_or(Error::<T>::InvalidResult)?;
+			let min_enactment_periods_u128 =
+				round_infos.time_periods.ok_or(Error::<T>::InvalidResult)?.min_enactment_period;
+			let min_enactment_periods =
+				min_enactment_periods_u128.try_into().map_err(|_| Error::<T>::InvalidResult)?;
+			let spend_valid_from =
+				round_infos.round_ending_block.saturating_add(min_enactment_periods);
+
+			let ref_index = infos.index;
+			let amount = infos.amount;
+			if let Some(ref_infos) = T::Governance::get_referendum_info(ref_index.into()) {
+				let state = T::Governance::handle_referendum_info(ref_infos);
+				match state {
+					Some(ReferendumStates::Approved) => {
+						let pot = Self::pot_account();
+						let balance = T::NativeBalance::balance(&pot);
+						let minimum_balance = T::NativeBalance::minimum_balance();
+						// check if the pot has enough fund for the Spend
+						// Check that the Pot as enough funds for the transfer
+						let remaining_balance = balance.saturating_sub(infos.amount);
+						ensure!(remaining_balance > minimum_balance, Error::<T>::NotEnoughFunds);
+						infos.spend_created = true;
+						WhiteListedProjectAccounts::<T>::mutate(project_id.clone(), |val| {
+							*val = Some(infos.clone())
+						});
+						// create a spend for project to be rewarded
+						let new_spend = SpendInfo::<T>::new(&infos, spend_valid_from);
+						Self::deposit_event(Event::ProjectFundingAccepted { project_id, amount });
+						Self::deposit_event(Event::SpendCreated {
+							amount: new_spend.amount,
+							project_id: infos.project_id.clone(),
+						});
+					},
+					Some(ReferendumStates::Rejected) =>
+						Self::deposit_event(Event::ProjectFundingRejected { project_id }),
+					Some(ReferendumStates::Ongoing) => (),
+					None => (),
+				}
+				Ok(())
+			} else {
+				Err(Error::<T>::ReferendumNotFound.into())
+			}
+		}
+
+		/// User's funds unlock
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// This extrinsic allows users to unlock funds related to a specific project,
+		/// provided the locking period (which is dependant of the conviction) has ended.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::NoVoteData`]: No vote data found for the specified project
+		///  
+		/// ## Events
+		#[pallet::call_index(7)]
+		#[pallet::weight(<T as Config>::WeightInfo::release_voter_funds(T::MaxProjects::get()))]
+		pub fn release_voter_funds(
```

Comment:
For the same reason, funds are handled by `pallet-conviction-voting`, no need to handle it in this pallet.